azure-keyvault/azure/keyvault/keys/_models.py

```diff
@@ -1,406 +1,219 @@
-        "tags": {"key": "tags", "type": "{str}"},
-        "managed": {"key": "managed", "type": "bool"},
-        "recovery_id": {"key": "recoveryId", "type": "str"},
-        "scheduled_purge_date": {"key": "scheduledPurgeDate", "type": "unix-time"},
-        "deleted_date": {"key": "deletedDate", "type": "unix-time"},
-    }
-
-    def __init__(self, **kwargs):
-        super(DeletedKeyItem, self).__init__(**kwargs)
-        self.recovery_id = kwargs.get("recovery_id", None)
-        self.scheduled_purge_date = None
-        self.deleted_date = None
+    @property
+    def key_material(self):
+        # type: () -> Dict[str, str]
+        return self._key_material
 
 
-class DeletedKeyItemPaged(Paged):
+class DeletedKey(KeyAttributes):
+    """A Deleted key consisting of its id, attributes, and tags, as
+    well as when it will be purged, if soft-delete is enabled for the vault.
     """
-    A paging container for iterating over a list of :class:`DeletedKeyItem <azure.keyvault.v7_0.models.DeletedKeyItem>` object
-    """
-
-    _attribute_map = {
-        "next_link": {"key": "nextLink", "type": "str"},
-        "current_page": {"key": "value", "type": "[DeletedKeyItem]"},
-    }
 
-    def __init__(self, *args, **kwargs):
-        super(DeletedKeyItemPaged, self).__init__(*args, **kwargs)
+    def __init__(self, attributes, vault_id, deleted_date=None, recovery_id=None, scheduled_purge_date=None, **kwargs):
+        # type: (models.KeyAttributes, str, Optional[datetime], Optional[str], Optional[datetime], Mapping[str, Any]) -> None
+        super(DeletedKey, self).__init__(attributes, vault_id, **kwargs)
+        self._deleted_date = deleted_date
+        self._recovery_id = recovery_id
+        self._scheduled_purge_date = scheduled_purge_date
+        # self._key_material = key # we don't expose the key i.e jsonWebKey just like we don't expose the value on deleted secret? why?
+
+    @classmethod
+    def _from_deleted_key_bundle(cls, deleted_key_bundle):
+        # type: (models.DeletedKeyBundle) -> DeletedKey
+        """Construct a DeletedKey from an autorest-generated DeletedKeyBundle"""
+        return cls(
+            deleted_key_bundle.attributes,
+            deleted_key_bundle.key.kid,
+            deleted_key_bundle.deleted_date,
+            deleted_key_bundle.recovery_id,
+            deleted_key_bundle.scheduled_purge_date,
+            managed=deleted_key_bundle.managed,
+            tags=deleted_key_bundle.tags,
+            # deleted_key_bundle.key, 
+            # TODO: deleted key item doesn't have the jsonWebKey.
+            # Option 1 - Is it fine if DeletedKey does not have the jsonWebKey property OR
+            # Option 2- Should we just have the other deletedKey properties such as deleted_date, recovery_id, scheduled_purge_date as None initialy on KeyATtributes class and
+            # have the "new" DeletedKey class to have the jsonWebKey + KeyAttributes properties
+        )
```

Comment:
@schaabs 

The `DeletedKey` model currently doesn't have the jsonWebKey property on it as `DeletedKeyItem` doesn't return one from the service.

Would it be a better idea to club together the deleted key properties with the `KeyAttributes` model class and have them initially as None or rather not have the jsonWebKey property on the `DeletedKey` model at all?